### PR TITLE
fix: SemVer compare functions not accepting mod objects

### PR DESCRIPTION
**Pull request by Enduriel**
Making a PR here because I haven't really used this approach where we have 1 shared verification function so wanna make sure it's fine with everyone.

### CODE REVIEW

File: scripts/config/msu/semver.nut [hunk 1]
```diff
@@ -99,25 +99,32 @@
...
+	{
+		::MSU.requireOneFromTypes(["string", "table", "instance"], _version1, _version2);
+		if (typeof _version1 == "instance") ::MSU.requireInstanceOf(::MSU.Class.Mod, _version1);
+		if (typeof _version2 == "instance") ::MSU.requireInstanceOf(::MSU.Class.Mod, _version2);
+	}
+
 	function compareVersionWithOperator( _version1, _operator, _version2 )
 	{
-		::MSU.requireOneFromTypes(["string", "table"], _version1, _version2);
+		this.verifyCompareInputs(_version1, _version2)
```
Comment (LordMidas) [inline scripts/config/msu/semver.nut:111]: semi-colon

File: scripts/config/msu/semver.nut [hunk 2]
```diff
@@ -99,25 +99,32 @@
...
 		if (typeof _version1 == "string") _version1 = this.getTable(_version1);
 		if (typeof _version2 == "string") _version2 = this.getTable(_version2);
 		return ::MSU.Utils.operatorCompare(this.compare(_version1, _version2), _operator)
 	}
 
 	function compareMajorVersionWithOperator( _version1, _operator, _version2 )
 	{
-		::MSU.requireOneFromTypes(["string", "table"], _version1, _version2);
+		this.verifyCompareInputs(_version1, _version2)
```
Comment (LordMidas) [inline scripts/config/msu/semver.nut:119]: semi-colon

File: scripts/config/msu/semver.nut [hunk 3]
```diff
@@ -99,25 +99,32 @@
...
 		if (typeof _version1 == "string") _version1 = this.getTable(_version1);
 		if (typeof _version2 == "string") _version2 = this.getTable(_version2);
 		return ::MSU.Utils.operatorCompare( _version1.Version[0] <=> _version2.Version[0], _operator);
 	}
 
 	function compareMinorVersionWithOperator( _version1, _operator, _version2 )
 	{
-		::MSU.requireOneFromTypes(["string", "table"], _version1, _version2);
+		this.verifyCompareInputs(_version1, _version2)
```
Comment (LordMidas) [inline scripts/config/msu/semver.nut:127]: semi-colon

File: scripts/config/msu/semver.nut [hunk 4]
```diff
@@ -99,25 +99,32 @@
 		return ret;
 	}
 
+	function verifyCompareInputs( _version1, _version2 ) // private
```
Comment (LordMidas) [inline scripts/config/msu/semver.nut:102]: Consider making a local function instead of a private one as this function is not accessed outside this file.

**Review by LordMidas — COMMENTED** [review]
Please see my comments. Looks good other than that.

File: scripts/config/msu/semver.nut (hunk 4 above)
Comment (Enduriel) [reply scripts/config/msu/semver.nut:102]: yeah local function is a good call

**Pull request merged**